Return correct result from module.run state

amendlik opened the pull request:
### What does this PR do?
Correct two bugs in the `module.run` state where the wrong result is returned.

1) When run in test mode, `module.run` should return `None`, indicating that changes would be made when not run in test mode
2) When no functions are passed to the state, it should return `False`, indicating an invocation error.

### What issues does this PR fix or reference?
None

### Previous Behavior
1) When run in test mode, `module.run` returned `True`, which indicates that no change would be made in in live mode. This is wrong because `modules.run` always executes the configured functions
2) When run with no functions passed, `module.run` returned `True`. This should return `False` to indicate an error.

### New Behavior
1) When run in test mode, `module.run` returns `None`, which indicates that changes will be made in in live mode. 
2) When run with no functions passed, `module.run` return `False`, which indicate an error.

### Merge requirements satisfied?
- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

amendlik committed:
Add tests for states.module

amendlik committed:
Return error if no functions a passed to module.run

krionbsd committed:
Fix pre-commit

Ch3LL commented on salt/states/module.py:
```diff
@@ -1,12 +1,11 @@
 
     There are two styles of calling ``module.run``. **The legacy style will no
-    longer be available starting in the 3005 release.** To opt-in early to the
+    longer be available starting in the 3001 release.** To opt-in early to the
```
this should remain 3005

amendlik replied:
Fixed.

Ch3LL replied:
I'm not seeing this resolved or the changelog entry. Also looks like pre-commit is failing now.

amendlik replied:
You're right - I didn't push the changes properly. It should be fixed now

Ch3LL reviewed (changes requested):
could you also add a changelog entry as documented here: https://docs.saltstack.com/en/master/topics/development/changelog.html

amendlik committed:
Create changelog for PR 58752

Ch3LL commented:
@amendlik looks like there are some merge conflicts that need to be cleaned up.

amendlik committed:
Return None from module.run state with test=true

module.run contained a bug where it would return True when run with
test=true. According to the documentation at
http://docs.saltstack.com/ref/states/writing.html#return-data
the state should return None in test mode if a change would be made
without test mode. Since module.run is not idempotent, the appropriate
return should always be None, as long as no errors were encountered.

amendlik commented:
@Ch3LL Ok, I rewrote my changes so they will merge with the updated files. What else do I need to do to get this merged?

amendlik commented:
@Ch3LL I would like to get this merged. What steps remain?

amendlik commented:
@Ch3LL I would like to get this merged. What steps remain?

amendlik commented:
@Ch3LL I would like to get this merged. What steps remain?

sagetherage committed:
Merge branch 'master' into module-run-state

sagetherage committed:
Merge branch 'master' into module-run-state

sagetherage commented:
re-running the failed windows test to see if it is flakey or not, will check back later today on results

Ch3LL committed:
Merge branch 'master' into module-run-state

s0undt3ch commented on tests/unit/states/test_module.py:
```diff
@@ -128,6 +128,34 @@ def test_run_module_not_available(self):
+        with patch.dict(module.__salt__, {}, clear=True), patch.dict(
+            module.__opts__, {"test": True, "use_superseded": ["module.run"]}
+        ):
+            ret = module.run(**{})
```
```suggestion
            ret = module.run()
```
Is `**{}` really necessary?

amendlik replied:
Good catch. I have removed that.

amendlik committed:
Remove unneeded empty kwargs dictionary
